core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -194,6 +197,64 @@ class AdminManager(val config: KafkaConfig,
     }
   }
 
+  def createPartitions(timeout: Int,
+                       newPartitions: Map[String, NewPartitions],
+                       validateOnly: Boolean,
+                       listenerName: ListenerName,
+                       callback: Map[String, ApiError] => Unit): Unit = {
+    var waiting = Map.empty[String, NewPartitions]
+    var done = Map.empty[String, ApiError]
+
+    newPartitions.foreach{case (topic, newPartition: NewPartitions) =>
+      try {
+        val oldNumPartitions = zkUtils.getTopicPartitionCount(topic) match {
+          case Some(count) => count
+          case None => throw new InvalidTopicException()
+        }
```

Comment:
Nit: `getOrElse` is a more concise version of this. Also, shouldn't this be `UnknownTopicOrPartition`?

Reply:
What is the purpose of `InvalidTopicException` then? Tell me and I'll update its javadoc.

Reply:
It's typically used for cases where the topic name is invalid. For example, creating a topic with a name that is not allowed. Or trying an operation on an internal topic.

Reply:
Btw, one difference is that `UnknownTopicOrPartitions` is retriable while `InvalidTopic` is not. So we should think about the impact of that. If someone creates a topic and soon after tries to add partitions, `UnknownTopicOrPartitions` would work better. However, for the common case, it seems that one would not want to retry if the topic doesn't exist.